src/Microsoft.ML.Data/DataView/CacheDataView.cs

```diff
@@ -605,8 +609,8 @@ private sealed class TrivialWaiter : IWaiter
 
             private TrivialWaiter(CacheDataView parent)
             {
-                Contracts.Assert(parent._rowCount >= 0);
-                _lim = parent._rowCount;
+                Contracts.Assert(parent.GetRowCount().HasValue);
+                _lim = parent.GetRowCount().Value;
```

Comment:
The multiple reads here is not incorrect, but having them means it takes more time to think about these lines. I would recommend using a single call to `GetRowCount()` for this method.